### PR TITLE
Fix display options not carrying over to DE

### LINKED ISSUE #1803: Use the same `display_options` for columns present in tables, for explorations
Also ensure the `display_options` persist when aggregating values

### PR DESCRIPTION
**Pull request by dmos62**
Related #1803

Didn't add a test, since change is trivial (and slightly pressed for time at this moment). Can add a test on request. Verified in local instance that display options do carry over from a table to a query.

## Checklist
 
- [x] My pull request has a descriptive title (not a vague title like `Update
  index.md`).
- [x] My pull request targets the `master` branch of the repository
- [x] My commit messages follow [best practices][best_practices].
- [x] My code follows the established code style of the repository.
- [ ] I added tests for the changes I made (if applicable).
- [x] I added or updated documentation (if applicable).
- [x] I tried running the project locally and verified that there are no
  visible errors.

[best_practices]:https://gist.github.com/robertpainsi/b632364184e70900af4ab688decf6f53

## Developer Certificate of Origin

<details>
<summary>Developer Certificate of Origin</summary>

```
Developer Certificate of Origin
Version 1.1

Copyright (C) 2004, 2006 The Linux Foundation and its contributors.
1 Letterman Drive
Suite D4700
San Francisco, CA, 94129

Everyone is permitted to copy and distribute verbatim copies of this
license document, but changing it is not allowed.


Developer's Certificate of Origin 1.1

By making a contribution to this project, I certify that:

(a) The contribution was created in whole or in part by me and I
    have the right to submit it under the open source license
    indicated in the file; or

(b) The contribution is based upon previous work that, to the best
    of my knowledge, is covered under an appropriate open source
    license and I have the right under that license to submit that
    work with modifications, whether created in whole or in part
    by me, under the same open source license (unless I am
    permitted to submit under a different license), as indicated
    in the file; or

(c) The contribution was provided directly to me by some other
    person who certified (a), (b) or (c) and I have not modified
    it.

(d) I understand and agree that this project and the contribution
    are public and that a record of the contribution (including all
    personal information I submit with it, including my sign-off) is
    maintained indefinitely and may be redistributed consistent with
    this project or the open source license(s) involved.
```

</details>

### CODE REVIEW

File: mathesar/models/query.py [hunk 1]
```diff
@@ -343,8 +340,31 @@ def _get_display_name_for_alias(self, alias):
...
+    @property
+    def _map_of_initial_col_alias_to_dj_column(self):
+        dj_column_ids = [col['id'] for col in self.initial_columns]
+        dj_columns = Column.objects.filter(pk__in=dj_column_ids)
+        initial_col_aliases = [
+            initial_col['alias']
+            for initial_col
+            in self.initial_columns
+        ]
+        return frozendict(
```
Comment (silentninja) [inline mathesar/models/query.py:362]: What is the reason to use a `frozendict` here?

**Review by silentninja — APPROVED** [review]
Looks good to me.

File: mathesar/models/query.py (hunk 1 above)
Comment (dmos62) [reply mathesar/models/query.py:362]: Just immutability. Could be converted into `@cached_property` without changes to the method body, for example.

**Review by pavish — DISMISSED** [review]
@dmos62 The issue #1803 mentions that this needs to be handled for aggregated items as well.

I can see that this works for normal columns, but display_options is not carried over for grouped columns (with same type), and for list aggregation, both of which are required.

I'm fine with handling them in a different PR though, if you'd like this to be merged right away, but they is essential for the initial release (and probably the demo video).

**Pull request merged**